evm_loader/cli/src/event_listener/listener_vm_tracer.rs

```diff
@@ -1,172 +1,62 @@
-use evm_loader::{
-    U256, Stack, ExitReason, Capture, StepTrace, StepResultTrace, SStoreTrace, SLoadTrace, Opcode
-};
-use crate::types::ec::trace::{INSTRUCTIONS, VMTracer};
-use super::vm_tracer::VmTracer;
-use log::{warn, info};
+use ethnum::U256;
+use evm_loader::evm::{Context, ExitStatus};
+use crate::types::ec::trace::{VMTracer, StorageDiff, MemoryDiff};
+use super::vm_tracer::{VmTracer};
 
 pub trait ListenerVmTracer {
-    fn step (&mut self, mes: &StepTrace);
-    fn step_result (&mut self, mes: &StepResultTrace);
-    fn sstore (&mut self, mes: &SStoreTrace);
-    fn sload (&mut self, mes: &SLoadTrace);
+    fn begin_vm(&mut self, context: Context, code: Vec<u8>);
+    fn end_vm(&mut self, status: ExitStatus);
```

Comment:
Same thing here, perhaps it makes sense to require `ListenerVmTracer` to implement `Drop` instead of having an explicit `end_vm`.

Reply:
See above